bot/discordbot.py

```diff
@@ -57,26 +57,35 @@
 # In order to avoid sending a user a message, you can pass None. This is helpfu for cases like the on_member_join function where it is not necessary
 # to send a message to the user as they are already joining the server.
 async def ensure_text_channel(entity, interaction: discord.Interaction, channel_name):
-    is_guild = isinstance(entity, discord.Guild)
-    if is_guild:
-        channel = discord.utils.get(entity.channels, name=channel_name)
-    else:
-        channel = discord.utils.get(entity.guild.channels, name=channel_name)
-    # Return the channel if it exists already
+    # Determine the correct context to search for or create the channel
+    guild = entity if isinstance(entity, discord.Guild) else entity.guild
+
+    # Attempt to find the channel in the guild
+    channel = discord.utils.get(guild.channels, name=channel_name)
     if channel:
-        return channel
+        return channel  # Return the channel if it already exists
+ 
     try:
-        if is_guild:
-            channel = await entity.create_text_channel(channel_name)
-        else:
-            channel = await entity.guild.create_text_channel(channel_name)
+        # Attempt to create the channel
+        channel = await guild.create_text_channel(channel_name)
         return channel
+    # Account for if the bot does not have the necessary permissions to create the channel
     except discord.errors.Forbidden:
-        message = "Reminder: The humanID Verification bot requires administrator permissions to create the get-verified channel and assign the humanID-Verified role. If you are the server admin, please ensure the bot has the necessary permissions to complete the verification process. If it does not, you may have to reinstall the bot with the correct permissions."
+        message = ("Reminder: The humanID Verification bot requires administrator permissions "
+                   "to create the get-verified channel and assign the humanID-Verified role. If"
+                    " you are the server admin, please ensure the bot has the necessary permissions"
+                    " to complete the verification process. If it does not, you may have to reinstall"
+                    " the bot with the correct permissions."
+                    )     
         if interaction:
             await interaction.response.send_message(message, ephemeral=True)
-        # No channel created
-        return
+        else:
+            # TODO: Is this the right behavior? How to account for cases if the bot does not have the necessary permissions
+            # and we lack an interaction to send a message to the user?
+            entity.send(message)
+
+        # TODO: What to return? No channel object available so we have to account for that in every other call
+        return None
```

Comment:
What is the correct behavior here? This is the case where there is no permissions AND no interaction. Do we send the user a Direct Message? If so what should it say? And what do we return? 

If we return None then we have to account for a None type everytime we do ```get_verified_channel = await ensure_text_channel()``` also this weakens the value of a "ensure" text channel function so I will possibly have to re-name it.

Reply:
@Lunasocute